File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.go
func lengthOfLIS(nums []int) int {
	n := len(nums)
	dp := make([]int, n)
	res := 0

	for i := 0; i < n; i++ {
		dp[i] = 1
		for j := 0; j < i; j++ {
			if nums[j] < nums[i] && dp[j]+1 > dp[i] {
				dp[i] = dp[j] + 1
			}
		}
		if dp[i] > res {
			res = dp[i]
		}
	}
	return res
}

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        res = []
        def dfs(perm, picked):
            if len(perm) == n:
                res.append(perm[:])
                return
            for i in range(n):
                if not picked[i]:
                    perm.append(nums[i])
                    picked[i] = True
                    dfs(perm, picked)
                    perm.pop()
                    picked[i] = False
        dfs([], [False]*n)
        return res